ext/gitsh/src/line_editor.c

```diff
@@ -0,0 +1,1973 @@
+/************************************************
```

Comment:
Does this file differ from upstream? Is it possible to clarify which is copied and which isn't? Or perhaps use e.g. `#include` to separate the files?